docs/data-sources/stream_processor.md

```diff
@@ -20,14 +20,68 @@ resource "mongodbatlas_stream_connection" "example-sample" {
   type            = "Sample"
 }
 
-resource "mongodbatlas_stream_processor" "stream-processor-example" {
+resource "mongodbatlas_stream_connection" "example-cluster" {
+  project_id      = var.project_id
+  instance_name   = mongodbatlas_stream_instance.example.instance_name
+  connection_name = "ClusterConnection"
+  type            = "Cluster"
+  cluster_name    = var.cluster_name
+  db_role_to_execute = {
+    role = "atlasAdmin"
+    type = "BUILT_IN"
+  }
+}
+
+resource "mongodbatlas_stream_connection" "example-kafka" {
+  project_id      = var.project_id
+  instance_name   = mongodbatlas_stream_instance.example.instance_name
+  connection_name = "KafkaPlaintextConnection"
+  type            = "Kafka"
+  authentication = {
+    mechanism = "PLAIN"
+    username  = var.kafka_username
+    password  = var.kafka_password
+  }
+  bootstrap_servers = "localhost:9092,localhost:9092"
+  config = {
+    "auto.offset.reset" : "earliest"
+  }
+  security = {
+    protocol = "PLAINTEXT"
+  }
+}
+
+resource "mongodbatlas_stream_processor" "stream-processor-sample-example" {
+  project_id     = var.project_id
+  instance_name  = mongodbatlas_stream_instance.example.instance_name
+  processor_name = "sampleProcessorName"
+  pipeline       = jsonencode([{ "$source" = { "connectionName" = resource.mongodbatlas_stream_connection.example-sample.connection_name } }, { "$emit" = { "connectionName" : "__testLog" } }])
+  state          = "CREATED"
+}
+
+resource "mongodbatlas_stream_processor" "stream-processor-cluster-example" {
   project_id     = var.project_id
   instance_name  = mongodbatlas_stream_instance.example.instance_name
-  processor_name = "processorName"
-  pipeline       = "[{\"$source\":{\"connectionName\":\"sample_stream_solar\"}},{\"$emit\":{\"connectionName\":\"__testLog\"}}]"
+  processor_name = "clusterProcessorName"
+  pipeline       = jsonencode([{ "$source" = { "connectionName" = resource.mongodbatlas_stream_connection.example-cluster.connection_name } }, { "$emit" = { "connectionName" : "__testLog" } }])
```

Comment:
[nit] would keep each stage on a separate line for readability (applies to all places)

Reply:
you mean the source and emit in different lines?